drivers/firmware/scmi/Kconfig

```diff
@@ -48,6 +48,19 @@ config ARM_SCMI_SHMEM_INIT_PRIORITY
 	help
 	  SCMI SHMEM driver device initialization priority.
 
+config ARM_SCMI_SHMEM_VENDOR_NXP
```

Comment:
I'd suggest making this more generic: `CONFIG_ARM_SCMI_NXP_VENDOR_EXTENSIONS`. Using this, we can have a distributed approach with our Kconfig and CMakeLists.txt files (i.e: each vendor can have its own Kconfig with their own vendor-specific configurations and a CMakeLists.txt file for adding source files based on their own configurations).